lib/index.js

```diff
@@ -170,14 +181,45 @@ export function createUnifiedLanguageServer({
   const documents = new TextDocuments(TextDocument)
   /** @type {Set<string>} */
   const workspaces = new Set()
+  /** @type {UnifiedLanguageServerSettings} */
+  const defaultSettings = {requireConfig: false}
+  let globalSettings = defaultSettings
+  /** @type {Map<string, Promise<UnifiedLanguageServerSettings>>} */
+  const documentSettings = new Map()
   let hasWorkspaceFolderCapability = false
+  let hasConfigurationCapability = false
+
+  /**
+   * @param {string} scopeUri
+   * @returns {Promise<UnifiedLanguageServerSettings>}
+   */
+  async function getDocumentSettings(scopeUri) {
+    if (!hasConfigurationCapability) {
+      return globalSettings
+    }
+    let result = documentSettings.get(scopeUri)
+    if (!result) {
+      result = connection.workspace
+        .getConfiguration({scopeUri, section: configurationSection})
+        .then((raw) => ({requireConfig: Boolean(raw.requireConfig)}))
+      documentSettings.set(scopeUri, result)
+    }
+    return result
+  }
 
   /**
    * @param {string} cwd
    * @param {VFile[]} files
    * @param {boolean} alwaysStringify
+   * @param {boolean} ignoreUnconfigured
+   * @returns {Promise<VFile[]>}
    */
-  async function processWorkspace(cwd, files, alwaysStringify) {
+  async function processWorkspace(
+    cwd,
+    files,
+    alwaysStringify,
+    ignoreUnconfigured
+  ) {
     /** @type {EngineOptions['processor']} */
     let processor
 
```

Comment:
Dropping a note to future self here: if `ignoreUnconfigured` is true, we shouldn’t require the user to install a processor.